src/ast/nodes/Identifier.ts

```diff
@@ -15,13 +15,29 @@ import LocalVariable from '../variables/LocalVariable';
 import Variable from '../variables/Variable';
 import * as NodeType from './NodeType';
 import SpreadElement from './SpreadElement';
-import { ExpressionEntity, LiteralValueOrUnknown } from './shared/Expression';
+import { ExpressionEntity, LiteralValueOrUnknown, UnknownValue } from './shared/Expression';
 import { ExpressionNode, NodeBase } from './shared/Node';
 import { PatternNode } from './shared/Pattern';
 
 export type IdentifierWithVariable = Identifier & { variable: Variable };
 
+const tdzInitTypesToIgnore = {
+	__proto__: null,
+	ArrowFunctionExpression: true,
+	ClassExpression: true,
+	FunctionExpression: true,
+	ObjectExpression: true
+};
+
+const variableKinds = {
+	__proto__: null,
+	const: true,
+	let: true,
+	var: true
+};
+
 export default class Identifier extends NodeBase implements PatternNode {
+	TDZ: boolean | undefined = undefined;
```

Comment:
following our naming conventions, this should not start with a capital letter, so probably rather `tdz`, or maybe better `isTDZAccess` to clarify it is a boolean.